File: leet_code/string/3/3.py
def lenght_of_longest_substring(s:str)->int:
    seen = set()
    left = right = max_len = 0
    while right < len(s):
        if s[right] not in seen:
            seen.add(s[right])
            max_len = max(max_len,right-left+1)
            right+=1
        else:
            seen.remove(s[left])
            left+=1

    return max_len

print(lenght_of_longest_substring("abcabcbb"))

File: leet_code/string/3/3.js
function longestSubString(s){
    let seen = new Set();
    let left = 0, right = 0, maxLen = 0;
    while(right<s.length){
        if(!seen.has(s[right])){
            seen.add(s[right]);
            maxLen = Math.max(maxLen,right-left+1);
            right++;
        }else{
            seen.delete(s[left]);
            left++;
        }
    }
    return maxLen;
}

console.log(longestSubString("abcabcbb"));